tsz-compress/src/v2/halfvec.rs

```diff
@@ -0,0 +1,424 @@
+            let len = word_lists.clone().map(|w| w.len).sum::<usize>() / 2;
+            let mut bytes = Vec::with_capacity(len + 1);
+
+            // Keep track of whether we are on the upper or lower nibble across word lists
+            let mut upper = true;
+            let mut byte = 0u8;
+
+            // Iterate over all the words in all the word lists
+            for word in word_lists.flat_map(|w| w.words.iter()) {
+                if upper {
+                    match word {
+                        HalfWord::Half(value) => {
+                            // Shift the value into the upper nibble
+                            byte = value << 4;
+                            // We are now on the lower nibble
+                            upper = false;
+                            // println!("upper nibble: {:b}", value);
+                        }
+                        HalfWord::Byte(value) => {
+                            // Use both nibbles from the byte
+                            known_append(&mut bytes, *value);
+                            // println!("upper byte: {:b}", value);
+                        }
+                        HalfWord::Full(value) => {
+                            // Use both nibbles from the top of the full
+                            known_append(&mut bytes, (value >> 24) as u8);
+                            // Use both nibbles from the top middle of the full
+                            known_append(&mut bytes, (value >> 16) as u8);
+                            // Use both nibbles from the bottom middle of the full
+                            known_append(&mut bytes, (value >> 8) as u8);
+                            // Use both nibbles from the bottom of the full
+                            known_append(&mut bytes, *value as u8);
+                            // println!("upper full: {:b}", value);
+                        }
+                    }
+                } else {
+                    match word {
+                        HalfWord::Half(value) => {
+                            // Fill the lower nibble, the upper nibble is already filled
+                            byte |= value & 0x0F;
+                            known_append(&mut bytes, byte);
+                            // We are now on the upper nibble
+                            upper = true;
+                            // println!("lower nibble: {:b}", value);
+                        }
+                        HalfWord::Byte(value) => {
+                            // Fill the lower nibble with the upper nibble of the value
+                            byte |= value >> 4;
+                            known_append(&mut bytes, byte);
+                            // Use the lower nibble from the value as the upper nibble
+                            byte = (value << 4) as u8;
+                            // We are still on the lower nibble
+                            // println!("lower byte: {:b}", value);
```

Comment:
we should get rid of commented out printlns

Reply:
Removed commented out println in commits: [71e143f](https://github.com/qsib-cbie/tsz/pull/45/commits/71e143fb79db34b659fbe859773cc39ee311f724) and [fe81490](https://github.com/qsib-cbie/tsz/pull/45/commits/fe814907d974d63ac0406c6ee9545d8fd7951a6b).